### PR TITLE
Add ESLint

### PR DESCRIPTION
**Pull request by uraniumanchor**
# Contributing to the Donation Tracker

First of all, thank you for taking the time to contribute!

Please fill out the template below and check the following boxes:

~~- [ ] I've added tests or modified existing tests for the change.~~
- [X] I've humanly end-to-end tested the change by running an instance of the tracker.

### Issue from Pivotal Tracker

https://www.pivotaltracker.com/story/show/169615168

### Description of the Change

Setting up ESLint (and by extension TSLint) rules and fixing up all the errors it found.

### Possible Drawbacks

The linter might have gotten something wrong?

### Verification Process

The only real code change I had to make was in a code block that can't even be accessed yet (it's supposed to allow adding new runs via the schedule editor, but there's no way to actually get in that state with the current app), but I ran through the pages just to make sure nothing seemed broken.

### CODE REVIEW

File: bundles/public/spinner.js [hunk 1]
```diff
@@ -7,7 +7,11 @@ class Spinner extends React.PureComponent {
   };
 
   render() {
-    return this.props.spinning ? <img src={STATIC_URL + this.props.imageFile} /> : <span>{this.props.children}</span>;
+    return this.props.spinning ? (
+      <img src={window.STATIC_URL + this.props.imageFile} />
+    ) : (
+      <span>{this.props.children}</span>
+    );
```
Comment (faultyserver) [inline bundles/public/spinner.js]: this seems to not match the prettier config, which uses 120 width, but the eslint config has `"max-len": 80`. I would prefer keeping 120, or compromising to have both at 100, but they definitely shouldn't conflict

File: package.json [hunk 2]
```diff
@@ -8,6 +8,8 @@
     "test": "karma start --single-run",
     "test:watch": "karma start",
     "fix:prettier": "prettier --write \"bundles/**\"",
+    "lint": "eslint bundles spec --ext js --ext jsx --ext ts --ext tsx --quiet",
+    "lint:fix": "yarn eslint --fix",
```
Comment (faultyserver) [inline package.json]: `fix:prettier` and `lint:fix` should use the same style. idc which, but imo they should match.

File: bundles/public/spinner.js (hunk 1 above)
Comment (uraniumanchor) [reply bundles/public/spinner.js]: Interestingly enough the rule is turned off despite having some options defined for it, so I'll just remove it from eslint. I forget where I lifted the base config from.

File: bundles/public/api/actions/models.js [hunk 3]
```diff
@@ -40,6 +39,7 @@ function onModelCollectionAdd(model, models) {
   };
 }
 
+// eslint-disable-next-line no-unused-vars
```
Comment (faultyserver) [inline bundles/public/api/actions/models.js]: what's it saying is unused here? the function? both `model` and `models` get referenced inside hm..

File: package.json [hunk 4]
```diff
@@ -8,6 +8,10 @@
     "test": "karma start --single-run",
     "test:watch": "karma start",
     "fix:prettier": "prettier --write \"bundles/**\"",
+    "fix:lint": "yarn lint --fix",
+    "fix:all": "yarn fix:prettier && yarn fix:lint",
+    "lint": "eslint bundles spec --ext js --ext jsx --ext ts --ext tsx --quiet",
+    "yarn:install": "yarn install",
```
Comment (faultyserver) [inline package.json]: is this...useful for something?

Would just be run as `yarn yarn:install` which seems kind of redundant lol.

File: bundles/public/api/actions/models.js (hunk 3 above)
Comment (uraniumanchor) [reply bundles/public/api/actions/models.js]: The function itself isn't used (yet).

File: package.json (hunk 4 above)
Comment (uraniumanchor) [reply package.json]: I have a task set up in PyCharm to start the webpack-dev-server, but if I want to make sure the deps are up to date I need to have it run a task first, and the least hoop jumping to force it to use yarn was this.

**Pull request merged**